src/main.rs

```diff
@@ -85,23 +85,20 @@ where
             }
             len => {
                 // Register the data that became available
-                used += len
+                used += len;
+                if let Some(last_delim_pos) = buf[..used].iter().rposition(|chr| *chr == delim) {
+                    buf[..last_delim_pos]
+                        .split(|chr| *chr == delim)
+                        .try_for_each(|line| handle_line(line))?;
```

Comment:
I am not quite happy with performing two passes on the rightmost data; finding the delimiter & copying it to the start…
This also accesses the data out of order; it probably doesn't have an actual impact in most real world scenarios but I do believe the cache coherency of this is slightly worse…

Although functionally it looks quite fine! :+1:

Reply:
I am not totally happy either. But I prefer combinator style because [`rayon`](https://docs.rs/crate/rayon/1.3.0) exists, and I can simply `use rayon::{split, try_for_each};` to get parallel computing.

Reply:
Oh I just found a even better way!

Reply:
Sure, but we're not going to use thread based parallelism for such a small bit of code, we're not using rayon and not doing something always beat doing it in parallel…so what is the purpose of this change?

Reply:
For the sake of combinator 😛  I prefer combinators because they give a name to each step and follow the principle of reusable code. I do not actually intend to merge this in, just showing another style.

I tried to find a way to not two-pass the right-most data but failed 😞 
I thought to use `peek()` and stop at one before last but the last `peek()` passes the data anyway.

Last resort I can make a new iterator called `split_without_last()` 🤔